### PR TITLE
Don't open a standalone editor during editing in a diff editor

### LINKED ISSUE #1218: During editing in a diff editor, a standalone editor for the same resource gets opened

### PR DESCRIPTION
**Pull request by akosyakov**
@JanKoehnlein please review, it fixes #1218

### CODE REVIEW

File: packages/monaco/src/browser/monaco-diff-editor.ts [hunk 1]
```diff
@@ -35,6 +35,7 @@ export class MonacoDiffEditor extends MonacoEditor {
         override?: IEditorOverrideServices,
     ) {
         super(new URI(''), modifiedModel, node, m2p, p2m, options, override);
+        this.documents.add(originalModel);
```
Comment (AlexTugarev) [inline packages/monaco/src/browser/monaco-diff-editor.ts]: @akosyakov why do we do the book-keeping here?

Comment (akosyakov) [reply packages/monaco/src/browser/monaco-diff-editor.ts]: look at `MonacoEditor.findByDocument`

File: packages/monaco/src/browser/monaco-editor.ts [hunk 2]
```diff
@@ -342,3 +305,46 @@ export class MonacoEditor implements TextEditor, IEditorReference {
...
+
+    export function get(editorWidget: EditorWidget | undefined) {
+        if (editorWidget && editorWidget.editor instanceof MonacoEditor) {
+            return editorWidget.editor;
+        }
+        return undefined;
+    }
+
+    export function findByDocument(manager: EditorManager, document: MonacoEditorModel): MonacoEditor[] {
+        return getAll(manager).filter(editor => editor.documents.has(document));
```
Comment (AlexTugarev) [inline packages/monaco/src/browser/monaco-editor.ts]: would the predicate `editor.document === document` work too?

Comment (akosyakov) [reply packages/monaco/src/browser/monaco-editor.ts]: no, diff editor has 2 documents

Comment (AlexTugarev) [reply packages/monaco/src/browser/monaco-editor.ts]: thanks! I got it. the other is added in super constructor.

**Pull request merged**